### PR TITLE
Rewrite Example to react-navigation v5

### PR DESCRIPTION
**Pull request by WoLewicki**
## Description

PR rewriting the Example app to `react-navigation v5`. As a side note, the `INTERACTABLE_SCREENS` are not used in the project, I am not sure if they should still be there and are not exposed on purpose so I kept it that way.

## Changes

Changed all `App.js` files to use navigators from newer version of `react-navigation`. Changed all `Header.HEIGHT` measures to usage of hook. Removed unnecessary libraries from `package.json`.

## Test code and steps to reproduce

Example App.

## Screenshots

iOS before and after:

image 
image

image
image

image
image

Android before and after:

image
image

image
image

image
image

## Checklist

- [x] Included code example that can be used to test this change
- [x] Ensured that CI passes

### CODE REVIEW

File: Example/src/LightboxExample.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ const styles = StyleSheet.create({
   container: {
     paddingTop: 0,
     height:
-      Platform.OS === 'web' ? dimensions.height - Header.HEIGHT : undefined,
+      Platform.OS === 'web' ? dimensions.height - useHeaderHeight() : undefined,
```
Comment (jakub-gonet) [inline Example/src/LightboxExample.js:41]: Will it work at top level?

File: Example/src/LightboxExample.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ const timingConfig = {
   easing: Easing.bezier(0.33, 0.01, 0, 1),
 };
 
-const HEADER_HEIGHT = Header.HEIGHT - StatusBar.currentHeight;
+let HEADER_HEIGHT = StatusBar.currentHeight;
```
Comment (jakub-gonet) [inline Example/src/LightboxExample.js:124]: This shouldn't be uppercased if it's not a constant. Couldn't it be moved to components?

File: Example/src/ScrollEventExample.js [hunk 3]
```diff
@@ -6,12 +6,16 @@ import Animated, {
   useAnimatedStyle,
   useAnimatedScrollHandler,
 } from 'react-native-reanimated';
-import { Header } from 'react-navigation-stack';
+import { useHeaderHeight } from '@react-navigation/stack';
 
-const windowHeight = Dimensions.get('window').height - Header.HEIGHT;
+let HEADER_HEIGHT = 0;
```
Comment (jakub-gonet) [inline Example/src/ScrollEventExample.js:11]: Ditto.

File: Example/src/ScrollableViewExample.js [hunk 4]
```diff
@@ -47,13 +47,15 @@ function friction(value) {
   return res;
 }
 
+let HEADER_HEIGHT = 0;
```
Comment (jakub-gonet) [inline Example/src/ScrollableViewExample.js:50]: Ditto.

File: Example/src/WebSpecific/LightBoxExample.js [hunk 5]
```diff
@@ -15,10 +15,12 @@ import {
   PanGestureHandler,
   TouchableWithoutFeedback,
 } from 'react-native-gesture-handler';
-import { Header } from 'react-navigation-stack';
+import { useHeaderHeight } from '@react-navigation/stack';
 
 const AnimatedImage = Animated.createAnimatedComponent(Image);
 
+let HEADER_HEIGHT = 0;
```
Comment (jakub-gonet) [inline Example/src/WebSpecific/LightBoxExample.js:22]: Ditto.

File: Example/src/App.js [hunk 6]
```diff
@@ -127,34 +117,46 @@ export function MainScreenItem({ item, onPressItem, screens }) {
...
-const ExampleApp = createSwitchNavigator({
-  Reanimated2App,
-  Reanimated1,
-});
+export default function App() {
+  const [useRea2, setUseRea2] = React.useState(true);
+
+  return (
+    <NavigationContainer>
+      {useRea2 ? Reanimated2(setUseRea2) : Reanimated1(setUseRea2)}
```
Comment (jakub-gonet) [inline Example/src/App.js:156]: Why do we need to specify if we're using Rea v1 or v2?

**Review by jakub-gonet — COMMENTED** [review]
Could you add some screenshots how it looks like now?

Have you tested the web version as well?

File: Example/src/App.js (hunk 6 above)
Comment (WoLewicki) [reply Example/src/App.js:156]: It is a state used to switch between Reanimated2 and Reanimated1 examples, something like in auth flow: https://reactnavigation.org/docs/auth-flow.

File: Example/src/LightboxExample.js (hunk 1 above)
Comment (WoLewicki) [reply Example/src/LightboxExample.js:41]: It's a hook so it is not possible. I am not sure if my current implementation is valid, but I don't know how to do it better atm.

File: Example/src/LightboxExample.js [hunk 7]
```diff
@@ -138,7 +119,7 @@ function ImageTransition({ activeImage, onClose }) {
 
   const targetX = useSharedValue(0);
   const targetY = useSharedValue(
-    (dimensions.height - targetHeight) / 2 - HEADER_HEIGHT
+    (dimensions.height - targetHeight) / 2 - useHeaderHeight()
```
Comment (jakub-gonet) [inline Example/src/LightboxExample.js:122]: Could you move it above and assign it to variable? I think hooks should be near the top of a component as a convention.

File: Example/src/ScrollEventExample.js [hunk 8]
```diff
@@ -40,12 +39,17 @@ function ScrollExample() {
     };
   });
 
+  const windowHeight = Dimensions.get('window').height - useHeaderHeight();
```
Comment (jakub-gonet) [inline Example/src/ScrollEventExample.js:42]: Same as above.

**Review by jakub-gonet — APPROVED** [review]

**Review by Szymon20000 — APPROVED** [review]

**Pull request merged**